### PR TITLE
Copyedit security-cors.adoc

### PR DESCRIPTION
**Pull request by rolfedh**
Improve style and clarity.

### CODE REVIEW

File: docs/src/main/asciidoc/security-cors.adoc [hunk 1]
```diff
@@ -4,68 +4,91 @@ and pull requests should be submitted there:
...
-:diataxis-type: concept
+:diataxis-type: reference
 :categories: security,web
-:keywords: cors,http
+:summary: This guide covers enabling and configuring the CORS filter, which uses HTTP headers to securely manage cross-origin browser requests.
+:keywords: cors,http,configuration, security, headers
 :extensions: io.quarkus:quarkus-vertx-http
 
-Cross-origin resource sharing (CORS) is an HTTP-header-based mechanism that allows a server to indicate any origins other than its own, from which a browser should permit loading resources.
+You can enable and configure the CORS filter, which uses HTTP headers to securely manage cross-origin browser requests.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-cors.adoc:15]: @rolfedh I'm not sure it is correct. The filter does not manage the cors requests securily with the help of headers, it is only supports a mechanism to indicate to the browser what can be a safe CORS request... I'm not sure starting the doc with the message what this filter does is better.

File: docs/src/main/asciidoc/security-cors.adoc [hunk 2]
```diff
@@ -4,68 +4,91 @@ and pull requests should be submitted there:
...
+To enforce CORS policies in your application, enable the Quarkus CORS filter by adding the following line to the `src/main/resources/application.properties` file:
 
 [source, properties]
 ----
 quarkus.http.cors=true
 ----
 
-When the filter is enabled and identifies an HTTP request as cross-origin, it will enforce the CORS policy.
-It will also add headers configured with the following properties before forwarding the request to its intended destination, like a servlet, Jakarta REST resource, or other endpoints.
+The filter, implemented using the `jakarta.servlet.Filter` interface, intercepts all incoming HTTP requests to identify cross-origin requests and apply the configured policies.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-cors.adoc:32]:
```suggestion
The filter intercepts all incoming HTTP requests to identify cross-origin requests and apply the configured policies.
```

File: docs/src/main/asciidoc/security-cors.adoc [hunk 3]
```diff
@@ -4,68 +4,91 @@ and pull requests should be submitted there:
...
+It then adds CORS headers to the HTTP response, informing browsers about allowed origins and access parameters.
+Finally, it forwards the response to the appropriate endpoint, such as a servlet or Jakarta REST resource.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-cors.adoc:34]:
```suggestion
For preflight requests, it returns HTTP response immediately.
For regular CORS requests, it denies an access with HTTP 403 status if the configured policy can not allow the request, or forward it to the original destination if the policy has been applied successfully.

```

File: docs/src/main/asciidoc/security-cors.adoc [hunk 4]
```diff
@@ -4,68 +4,91 @@ and pull requests should be submitted there:
...
+<1> Enables the CORS filter.
+<2> Specifies allowed origins, including a regular expression.
+<3> Lists allowed HTTP methods for cross-origin requests.
+<4> Declares custom headers that clients can include in requests.
+<5> Identifies response headers that clients can access.
+<6> Sets how long preflight request results are cached.
+<7> Allows cookies or credentials in cross-origin requests.
 
-[NOTE]
+The `/https://([a-z0-9\\-_]+)\\\\.app\\\\.mydomain\\\\.com/` pattern functions as a regular expression due to the enclosing forward slashes.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-cors.adoc:63]:
```suggestion

```

**Review by rolfedh — COMMENTED** [review]
Thank you, @sberyozkin. Please let me know if we can make further improvements!

File: docs/src/main/asciidoc/security-cors.adoc [hunk 5]
```diff
@@ -4,68 +4,91 @@ and pull requests should be submitted there:
...
 ----
 quarkus.http.cors=true
 ----
 
-When the filter is enabled and identifies an HTTP request as cross-origin, it will enforce the CORS policy.
-It will also add headers configured with the following properties before forwarding the request to its intended destination, like a servlet, Jakarta REST resource, or other endpoints.
+The filter intercepts all incoming HTTP requests to identify cross-origin requests and applies the configured policy.
+It then adds CORS headers to the HTTP response, informing browsers about allowed origins and access parameters.
+For preflight requests, it returns HTTP response immediately.
+For regular CORS requests, it denies an access with HTTP 403 status if the configured policy can not allow the request, or forward it to the original destination if the policy has been applied successfully.
```
Comment (rolfedh) [inline docs/src/main/asciidoc/security-cors.adoc:35]: @sberyozkin Would  it be okay to say:
```suggestion
For regular CORS requests, it denies access with an HTTP 403 status if the request violates the configured policy or forwards it to the destination if the policy allows it.
```

**Comment by gsmet** [comment]
Didn't we already discuss the fact that we wanted to avoid large backports to 3.15 after the first release, included in doc?

Is there any particular reason why this PR has the `backport-3.15` label?

**Comment by rolfedh** [comment]
Hi @gsmet,

Could you please also consider backporting this change to 3.15?

Thank you!

**Review by sberyozkin — APPROVED** [review]
Thanks @rolfedh for improving the doc

**Comment by sberyozkin** [comment]
@rolfedh Hi Rolfe, PR has a merge commit: `[sha=bae368c86a0cfa7b361e36557d05415b5fe4f511 ; message="Merge branch 'main' into copyedit-cors-ref-topic"]`

**Pull request merged**